engines/alg/game_bountyhunter.cpp

```diff
@@ -0,0 +1,1343 @@
+	outSaveFile->writeUint16LE(currentSceneNum);
+	for (uint8 i = 0; i < 2; i++) {
+		outSaveFile->writeByte(_playerLives[i]);
+		outSaveFile->writeByte(_playerShots[i]);
+		outSaveFile->writeUint32LE(_playerScore[i]);
+	}
+	outSaveFile->writeByte(_unk_2ADA6);
+	outSaveFile->writeByte(_numPlayers);
+	outSaveFile->finalize();
+	delete outSaveFile;
+	return true;
+}
+
+bool GameBountyHunter::_LoadState() {
+	Common::InSaveFile *inSaveFile;
+	Common::String saveFileName = _vm->getSaveStateName(0);
+	if (!(inSaveFile = g_system->getSavefileManager()->openForLoading(saveFileName))) {
+		debug("Can't load file '%s', game not loaded", saveFileName.c_str());
+		return false;
+	}
+	uint32 header = inSaveFile->readUint32BE();
+	if (header != MKTAG('A', 'L', 'G', 'S')) {
+		warning("Unkown save file, header: %d", header);
+		return false;
+	}
+	inSaveFile->skip(1); // version, unused for now
+	_currentLevel = inSaveFile->readByte();
+	_currentSubLevelSceneId = inSaveFile->readUint16LE();
+	_continuesUsed = inSaveFile->readByte();
+	_restartScene = inSaveFile->readUint16LE();
+	for (uint8 i = 0; i < 2; i++) {
+		_playerLives[i] = inSaveFile->readByte();
+		_playerShots[i] = inSaveFile->readByte();
+		_playerScore[i] = inSaveFile->readUint32LE();
+	}
+	_unk_2ADA6 = inSaveFile->readByte();
+	_numPlayers = inSaveFile->readByte();
+	delete inSaveFile;
+	_gameInProgress = true;
+	return true;
+}
+
+// misc game functions
+void GameBountyHunter::_SetNextScene(uint16 sceneId) {
+	_cur_scene = Common::String::format("%d", sceneId);
+}
+
+void GameBountyHunter::_DisplayShotFiredImage(Common::Point *point) {
+	if (point->x >= _videoPosX && point->x <= (_videoPosX + _videoDecoder->getWidth()) && point->y >= _videoPosY && point->y <= (_videoPosY + _videoDecoder->getHeight())) {
+		uint16 targetX = point->x - _videoPosX;
+		uint16 targetY = point->y - _videoPosY;
+		AlgGraphics::drawImageCentered(_videoDecoder->getVideoFrame(), &_bulletholeIcon, targetX, targetY);
+	}
+}
+
+void GameBountyHunter::_EnableVideoFadeIn() {
+	// TODO implement
+}
+
+void GameBountyHunter::_IconShotgun() {
+	// TODO implement
+}
+
+void GameBountyHunter::_IconReset() {
+	// TODO implement
+}
+
+uint16 GameBountyHunter::_BeginLevel(uint8 levelNumber) {
+	_currentLevel = levelNumber;
+	_numSubLevelsDone = 0;
+	int index = (levelNumber * 24) + (_numLevelsDone * 6) + _numSubLevelsDone;
+	uint8 subLevel = _subLevelOrder[index];
+	uint16 sceneIndex = (_currentLevel * 5) + subLevel;
+	uint16 sceneNum = _subLevelSceneIds[sceneIndex];
+	_currentSubLevelSceneId = sceneNum;
+	return sceneNum;
+}
+
+uint16 GameBountyHunter::_RandomUnusedInt(uint8 max, uint16 *mask, uint16 exclude) {
+	if (max == 1) {
+		return 0;
+	}
+	// reset mask if full
+	uint16 fullMask = 0xFFFF >> (16 - max);
+	if (*mask == fullMask) {
+		*mask = 0;
+	}
+	uint16 random = 0;
+	// find an unused random number
+	while (1) {
+		random = _rnd->getRandomNumber(max - 1);
+		// check if bit is already used
+		unsigned int bit = 1 << random;
```

Comment:
```suggestion
		uint bit = 1 << random;
```
And please do mass-replace.

Reply:
fixed